kit/src/elements/button/mod.rs

```diff
@@ -135,15 +130,15 @@ pub fn Button<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
                         }
                     )
                 }
-                if let Some(_icon) = cx.props.icon {
+                if show_icon {
                     rsx!(
                         // for props, copy the defaults passed in by IconButton
                         common::icons::Icon {
                             ..common::icons::IconProps {
-                                class: None,
+                                class: if cx.props.loading.unwrap_or(false) {Some("spin-container-for-button")} else {None},
```

Comment:
```suggestion
                                class: if cx.props.loading.unwrap_or_default() {Some("spin-container-for-button")} else {None},
```

Reply:
We would instead take advantage of `then` so if `loading` is true, it would provide `Option::Some` otherwise it would be `Option::None`

```suggestion
                                class: if cx.props.loading.unwrap_or_default().then(||"spin-container-for-button"),
```